### ISSUE TITLE
ViT + CLIP

### ISSUE THREAD

**Issue by gboduljak** [ml-explore/mlx-examples#143]
Would it be worth implementing ViT and CLIP example?

**Comment by awni** [comment]
Yea that ones on our list of examples to add!

**Comment by gboduljak** [comment]
> Yea that ones on our list of examples to add! Are you interested in contributing it? If so which model would you use?

I would like to contribute :) I would use models from the official CLIP repository: https://github.com/openai/CLIP. If you have an alternative idea, please let me know.

**Comment by nkasmanoff** [comment]
@gboduljak I submitted a PR to your existing PR, which creates a local implementation of the CLIPImageProcessor. https://github.com/gboduljak/mlx-examples/pull/1

This should eliminate the dependency on transformers, aside from using it for downloading the model & tokenizer.

**Comment by gboduljak** [comment]
@nkasmanoff Thanks for the help. I will take a look at your work now.

**Comment by gboduljak** [comment]
@nkasmanoff I merged your PR, corrected the nits and I refactored your implementation so that everything is in ``preprocessing`` folder. Many thanks for the help. In future, we might drop this 'copy-paste' implementation from HuggingFace. Ideally, we should use [mlx-data](https://github.com/ml-explore/mlx-data). If you have time, it would be awesome to have ``mlx-data`` implementation of ``CLIPImageProcessor``.
